bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/ReadSessionCreator.java

```diff
@@ -144,7 +144,7 @@ public ReadSessionResponse create(
                         .setTable(tablePath)
                         .build())
                 .setMaxStreamCount(maxStreamCount)
-                .setPreferredMinStreamCount(preferredMinStreamCount)
+                .setPreferredMinStreamCount(Math.min(preferredMinStreamCount, maxStreamCount))
```

Comment:
We need to at least warn the user - please add an `if` prior to the call, and add a warning log that the preferredMinStreamCount is larger than the max

Reply:
Addressed.